### ISSUE TITLE
org.elasticsearch.transport.local.SimpleLocalTransportTests.testSendRandomRequests fails on 5.x

### ISSUE THREAD

**Issue by colings86** [elastic/elasticsearch#22784]
example failure:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.x+java9-periodic/1503/
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.x+multijob-intake/642

reproduce command (reproduces consistently):
```
gradle :core:test -Dtests.seed=175EAB60866BD0C5 -Dtests.class=org.elasticsearch.transport.local.SimpleLocalTransportTests -Dtests.method="testSendRandomRequests" -Dtests.security.manager=true -Dtests.locale=ar-BH -Dtests.timezone=America/Edmonton
```

Stack trace:
```
 2> WARNING: Uncaught exception in thread: Thread[elasticsearch[local_transport][T#1],5,TGRP-SimpleLocalTransportTests]
  2> java.lang.AssertionError
  1> [2017-01-25T03:53:14,708][INFO ][o.e.t.l.SimpleLocalTransportTests] [testNotifyOnShutdown]: after test
  2> 	at __randomizedtesting.SeedInfo.seed([175EAB60866BD0C5]:0)
  2> 	at org.junit.Assert.fail(Assert.java:86)
  1> [2017-01-25T03:53:14,709][INFO ][o.e.t.l.SimpleLocalTransportTests] [testSendRandomRequests]: before test
  1> [2017-01-25T03:53:14,721][INFO ][o.e.t.t.MockTransportService] [TS_A] publish_address {local[33]}, bound_addresses {local[33]}
  2> 	at org.junit.Assert.assertTrue(Assert.java:41)
  2> 	at org.junit.Assert.assertNotNull(Assert.java:621)
  2> 	at org.junit.Assert.assertNotNull(Assert.java:631)
  2> 	at org.elasticsearch.transport.AbstractSimpleTransportTestCase$1TestResponseHandler.handleException(AbstractSimpleTransportTestCase.java:1714)
  1> [2017-01-25T03:53:14,723][INFO ][o.e.t.t.MockTransportService] [TS_B] publish_address {local[34]}, bound_addresses {local[34]}
  2> 	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1032)
  2> 	at org.elasticsearch.transport.local.LocalTransport.handleException(LocalTransport.java:455)
  1> [2017-01-25T03:53:14,747][INFO ][o.e.t.t.MockTransportService] publish_address {local[35]}, bound_addresses {local[35]}
  2> 	at org.elasticsearch.transport.local.LocalTransport.handleResponseError(LocalTransport.java:446)
  2> 	at org.elasticsearch.transport.local.LocalTransport.processReceivedMessage(LocalTransport.java:326)
  2> 	at org.elasticsearch.transport.local.LocalTransport.lambda$receiveMessage$0(LocalTransport.java:288)
  2> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:527)
  2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  2> 	at java.lang.Thread.run(Thread.java:745)
  2> ينا 25, 2017 3:53:15 ص com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
  2> WARNING: Uncaught exception in thread: Thread[elasticsearch[TS_A][local_transport][T#1],5,TGRP-SimpleLocalTransportTests]
  2> java.lang.AssertionError
  2> 	at __randomizedtesting.SeedInfo.seed([175EAB60866BD0C5]:0)
  2> 	at org.junit.Assert.fail(Assert.java:86)
  2> 	at org.junit.Assert.assertTrue(Assert.java:41)
  2> 	at org.junit.Assert.assertNotNull(Assert.java:621)
  2> 	at org.junit.Assert.assertNotNull(Assert.java:631)
  2> 	at org.elasticsearch.transport.AbstractSimpleTransportTestCase$1TestResponseHandler.handleException(AbstractSimpleTransportTestCase.java:1714)
  2> 	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1032)
  2> 	at org.elasticsearch.transport.local.LocalTransport.handleException(LocalTransport.java:455)
  2> 	at org.elasticsearch.transport.local.LocalTransport.handleResponseError(LocalTransport.java:446)
  2> 	at org.elasticsearch.transport.local.LocalTransport.processReceivedMessage(LocalTransport.java:326)
  2> 	at org.elasticsearch.transport.local.LocalTransport.lambda$receiveMessage$0(LocalTransport.java:288)
  2> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:527)
  2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  2> 	at java.lang.Thread.run(Thread.java:745)
```

**Comment by cbuescher** [comment]
There is another one here today: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.x+dockeralpine-periodic/716

Reproduces on 5d48757db0b9d301869198fcd9dc3178a607a4a5 with the following 
```
gradle :core:test -Dtests.seed=B3894BBAFA14435E -Dtests.class=org.elasticsearch.transport.local.SimpleLocalTransportTests -Dtests.method="testSendRandomRequests" -Dtests.security.manager=true -Dtests.locale=sr-Latn-RS -Dtests.timezone=America/Maceio
```

**Comment by s1monw** [comment]
closed by 1ce8c01
